system/Commands/Generators/ModelGenerator.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+/**
+ * This file is part of the CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\Commands\Generators;
+
+use CodeIgniter\CLI\BaseCommand;
+use CodeIgniter\CLI\CLI;
+use CodeIgniter\CLI\GeneratorTrait;
+
+/**
+ * Generates a skeleton Model file.
+ */
+class ModelGenerator extends BaseCommand
+{
+	use GeneratorTrait;
+
+	/**
+	 * The Command's Group
+	 *
+	 * @var string
+	 */
+	protected $group = 'Generators';
+
+	/**
+	 * The Command's Name
+	 *
+	 * @var string
+	 */
+	protected $name = 'make:model';
+
+	/**
+	 * The Command's Description
+	 *
+	 * @var string
+	 */
+	protected $description = 'Generates a new model file.';
+
+	/**
+	 * The Command's Usage
+	 *
+	 * @var string
+	 */
+	protected $usage = 'make:model <name> [options]';
+
+	/**
+	 * The Command's Arguments
+	 *
+	 * @var array
+	 */
+	protected $arguments = [
+		'name' => 'The model class name.',
+	];
+
+	/**
+	 * The Command's Options
+	 *
+	 * @var array
+	 */
+	protected $options = [
+		'--table'     => 'Supply a table name.',
+		'--dbgroup'   => 'Database group to use. Defaults: "default".',
+		'--return'    => 'Return type, Options: [array, object, entity], Defaults: "array".',
+		'--namespace' => 'Set root namespace. Defaults: "APP_NAMESPACE".',
+		'--suffix'    => 'Append the component title to the class name (e.g. User => UserModel).',
+		'--force'     => 'Force overwrite existing file.',
+	];
+
+	/**
+	 * Actually execute a command.
+	 *
+	 * @param array $params
+	 */
+	public function run(array $params)
+	{
+		$this->component = 'Model';
+		$this->directory = 'Models';
+		$this->template  = 'model.tpl.php';
+
+		$this->execute($params);
+	}
+
+	/**
+	 * Prepare options and do the necessary replacements.
+	 *
+	 * @param string $class
+	 *
+	 * @return string
+	 */
+	protected function prepare(string $class): string
+	{
+		$table   = $this->getOption('table');
+		$DBGroup = $this->getOption('dbgroup');
+		$return  = $this->getOption('return') ?? 'array';
+
+		$table   = is_string($table) ? $table : strtolower($this->params[0] ?? CLI::getSegment(2));
+		$DBGroup = is_string($DBGroup) ? $DBGroup : 'default';
+
+		if (! in_array($return, ['array', 'object', 'entity'], true))
+		{
+			$return = CLI::prompt(lang('CLI.generator.returnType'), ['array', 'object', 'entity'], 'required'); // @codeCoverageIgnore
+			CLI::newLine();
+		}
+
+		if ($return === 'entity')
+		{
+			$return = str_replace('Models', 'Entities', $class);
+
+			if (strpos($return, 'Model'))
+			{
+				$return = str_replace('Model', 'Entity', $return);
+			}
```

Comment:
You are forcing here the suffix option. At least use the default setting of no suffixes.
```suggestion
			if ($pos = strrpos($return, 'Model'))
			{
				// Strip 'Model' from name
				$return = substr($return, 0, $pos);
			}
```

Reply:
your suggestion will not append `Entity` string to $returnType property, but the entity name will be suffixed with Entity, i will try to find another way to avoid this issue

Reply:
The suffixing of the Entity via the model generator should be configured by the `--suffix` option too.